### PR TITLE
CppCheck HVACMultiSpeedHeatPump

### PR DESCRIPTION
**Pull request by rraustad**
Pull request overview
---------------------
 - Fixes CppCheck issues
 - DESCRIBE PURPOSE OF THIS PULL REQUEST

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/HVACMultiSpeedHeatPump.cc [hunk 1]
```diff
@@ -538,12 +471,6 @@ namespace HVACMultiSpeedHeatPump {
         int i;                         // Index to speeds
         int j;                         // Index to speeds
         bool Found;                    // Flag to find autosize
-        int HeatingCoilInletNode;      // Heating coil inlet node number
-        int HeatingCoilOutletNode;     // Heating coil outlet node number
-        int CoolingCoilInletNode;      // Cooling coil inlet node number
-        int CoolingCoilOutletNode;     // Cooling coil outlet node number
-        int SuppHeatCoilInletNode;     // Supplemental heating coil inlet node number
-        int SuppHeatCoilOutletNode;    // Supplemental heating coil outlet node number
```
Comment (rraustad) [inline src/EnergyPlus/HVACMultiSpeedHeatPump.cc:546]: [src/EnergyPlus/HVACMultiSpeedHeatPump.cc:541]:(style),[variableScope],The scope of the variable 'HeatingCoilInletNode' can be reduced.
[src/EnergyPlus/HVACMultiSpeedHeatPump.cc:542]:(style),[variableScope],The scope of the variable 'HeatingCoilOutletNode' can be reduced.
[src/EnergyPlus/HVACMultiSpeedHeatPump.cc:543]:(style),[variableScope],The scope of the variable 'CoolingCoilInletNode' can be reduced.
[src/EnergyPlus/HVACMultiSpeedHeatPump.cc:544]:(style),[variableScope],The scope of the variable 'CoolingCoilOutletNode' can be reduced.
[src/EnergyPlus/HVACMultiSpeedHeatPump.cc:545]:(style),[variableScope],The scope of the variable 'SuppHeatCoilInletNode' can be reduced.
[src/EnergyPlus/HVACMultiSpeedHeatPump.cc:546]:(style),[variableScope],The scope of the variable 'SuppHeatCoilOutletNode' can be reduced.

File: src/EnergyPlus/HVACMultiSpeedHeatPump.cc [hunk 2]
```diff
@@ -1944,80 +1874,54 @@ namespace HVACMultiSpeedHeatPump {
...
-        using SteamCoils::SimulateSteamCoilComponents;
-        using WaterCoils::GetCoilMaxWaterFlowRate;
-        using WaterCoils::SimulateWaterCoilComponents;
-
         // SUBROUTINE LOCAL VARIABLE DECLARATIONS:
         static constexpr std::string_view RoutineName("InitMSHeatPump");
-        int InNode;     // Inlet node number in MSHP loop
-        int OutNode;    // Outlet node number in MSHP loop
-        int ZoneInNode; // Zone inlet node number in the controlled zone for MSHP
-        Real64 RhoAir;  // Air density at InNode
```
Comment (rraustad) [inline src/EnergyPlus/HVACMultiSpeedHeatPump.cc:1969]: [src/EnergyPlus/HVACMultiSpeedHeatPump.cc:1968]:(style),[variableScope],The scope of the variable 'ZoneInNode' can be reduced.

File: src/EnergyPlus/HVACMultiSpeedHeatPump.cc [hunk 3]
```diff
@@ -1944,80 +1874,54 @@ namespace HVACMultiSpeedHeatPump {
...
+        Real64 DeltaMassRate; // Difference of mass flow rate between inlet node and system outlet node
 
-        int ZoneInSysIndex(0);                            // number of zone inlet nodes counter in an airloop
         int NumAirLoopZones(0);                           // number of zone inlet nodes in an air loop
-        int ZoneInletNodeNum(0);                          // zone inlet nodes node number
         Real64 SumOfMassFlowRateMax(0.0);                 // the sum of mass flow rates at inlet to zones in an airloop
         Real64 CntrlZoneTerminalUnitMassFlowRateMax(0.0); // Maximum mass flow rate through controlled zone terminal unit
-        bool errFlag;
-        Real64 rho;    // local fluid density
-        Real64 MdotHR; // local temporary for heat recovery fluid mass flow rate (kg/s)
```
Comment (rraustad) [inline src/EnergyPlus/HVACMultiSpeedHeatPump.cc:1986]: [src/EnergyPlus/HVACMultiSpeedHeatPump.cc:1981]:(style),[variableScope],The scope of the variable 'ZoneInletNodeNum' can be reduced.
[src/EnergyPlus/HVACMultiSpeedHeatPump.cc:1984]:(style),[variableScope],The scope of the variable 'errFlag' can be reduced.

File: src/EnergyPlus/HVACMultiSpeedHeatPump.cc [hunk 4]
```diff
@@ -1944,80 +1874,54 @@ namespace HVACMultiSpeedHeatPump {
...
         Real64 ZoneLoadToCoolSPSequenced;
         Real64 ZoneLoadToHeatSPSequenced;
 
         bool ErrorsFound(false);        // flag returned from mining call
-        int SteamIndex(0);              // index of steam quality for steam heating coil
         Real64 mdot(0.0);               // local temporary for mass flow rate (kg/s)
         Real64 SteamDensity(0.0);       // density of steam at 100C, used for steam heating coils
         Real64 CoilMaxVolFlowRate(0.0); // coil fluid maximum volume flow rate
         Real64 QActual(0.0);            // coil actual capacity
-        int CoilAvailSchPtr(0);         // DX coil availability schedule pointer
```
Comment (rraustad) [inline src/EnergyPlus/HVACMultiSpeedHeatPump.cc:1996]: [src/EnergyPlus/HVACMultiSpeedHeatPump.cc:1996]:(style),[variableScope],The scope of the variable 'CoilAvailSchPtr' can be reduced.
[src/EnergyPlus/HVACMultiSpeedHeatPump.cc:1996]:(style),[unreadVariable],Variable 'CoilAvailSchPtr' is assigned a value that is never used.

File: src/EnergyPlus/HVACMultiSpeedHeatPump.cc [hunk 5]
```diff
@@ -2167,10 +2074,10 @@ namespace HVACMultiSpeedHeatPump {
         if (allocated(state.dataZoneEquip->ZoneEquipConfig) && MSHeatPump(MSHeatPumpNum).MyCheckFlag) {
             int zoneNum = MSHeatPump(MSHeatPumpNum).ControlZoneNum;
             int zoneInlet = MSHeatPump(MSHeatPumpNum).ZoneInletNode;
-            int coolingPriority = 0;
-            int heatingPriority = 0;
```
Comment (rraustad) [inline src/EnergyPlus/HVACMultiSpeedHeatPump.cc:2171]: [src/EnergyPlus/HVACMultiSpeedHeatPump.cc:2170]:(style),[variableScope],The scope of the variable 'coolingPriority' can be reduced.
[src/EnergyPlus/HVACMultiSpeedHeatPump.cc:2171]:(style),[variableScope],The scope of the variable 'heatingPriority' can be reduced.

File: src/EnergyPlus/HVACMultiSpeedHeatPump.cc [hunk 6]
```diff
@@ -4550,42 +4234,19 @@ namespace HVACMultiSpeedHeatPump {
...
-        // DERIVED TYPE DEFINITIONS
-        // na
-
         // SUBROUTINE LOCAL VARIABLE DECLARATIONS:
         Real64 QCoilActual;     // actual heating load met
         Real64 mdot;            // heating coil steam or hot water mass flow rate
         Real64 MinWaterFlow;    // coil minimum hot water mass flow rate, kg/s
         Real64 MaxHotWaterFlow; // coil maximum hot water mass flow rate, kg/s
         Real64 HotWaterMdot;    // actual hot water mass flow rate
-        int SolFlag;
```
Comment (rraustad) [inline src/EnergyPlus/HVACMultiSpeedHeatPump.cc:4588]: [src/EnergyPlus/HVACMultiSpeedHeatPump.cc:4588]:(style),[variableScope],The scope of the variable 'SolFlag' can be reduced.

File: src/EnergyPlus/HVACMultiSpeedHeatPump.cc [hunk 7]
```diff
@@ -3337,36 +3229,36 @@ namespace HVACMultiSpeedHeatPump {
...
             // heating
-            if (QZnReq > SmallLoad && QZnReq < FullOutput) {
+            if (QZnReq > DataHVACGlobals::SmallLoad && QZnReq < FullOutput) {
                 CalcMSHeatPump(
                     state, MSHeatPumpNum, FirstHVACIteration, CompressorOp, 1, 0.0, 0.0, TempOutput0, QZnReq, OnOffAirFlowRatio, SupHeaterLoad);
 
-                for (int i = 1; i <= MSHeatPump(MSHeatPumpNum).NumOfSpeedHeating; ++i) {
-                    if (i == 1) {
+                for (int k = 1; k <= MSHeatPump.NumOfSpeedHeating; ++k) {
+                    if (k == 1) {
```
Comment (rraustad) [inline src/EnergyPlus/HVACMultiSpeedHeatPump.cc:3155]: [src/EnergyPlus/HVACMultiSpeedHeatPump.cc:3349]:(style),[shadowVariable],Local variable 'i' shadows outer variable

File: src/EnergyPlus/HVACMultiSpeedHeatPump.cc [hunk 8]
```diff
@@ -3410,30 +3301,30 @@ namespace HVACMultiSpeedHeatPump {
...
 
             // Coolling
-            if (QZnReq < (-1.0 * SmallLoad) && QZnReq > FullOutput) {
+            if (QZnReq < (-1.0 * DataHVACGlobals::SmallLoad) && QZnReq > FullOutput) {
                 CalcMSHeatPump(
                     state, MSHeatPumpNum, FirstHVACIteration, CompressorOp, 1, 0.0, 0.0, TempOutput0, QZnReq, OnOffAirFlowRatio, SupHeaterLoad);
-                for (int i = 1; i <= MSHeatPump(MSHeatPumpNum).NumOfSpeedCooling; ++i) {
-                    if (i == 1) {
+                for (int k = 1; k <= MSHeatPump.NumOfSpeedCooling; ++k) {
+                    if (k == 1) {
```
Comment (rraustad) [inline src/EnergyPlus/HVACMultiSpeedHeatPump.cc:3221]: [src/EnergyPlus/HVACMultiSpeedHeatPump.cc:3416]:(style),[shadowVariable],Local variable 'i' shadows outer variable

File: src/EnergyPlus/HVACMultiSpeedHeatPump.cc [hunk 9]
```diff
@@ -2836,53 +2748,44 @@ namespace HVACMultiSpeedHeatPump {
...
-
-        using PlantUtilities::RegisterPlantCompDesignFlow;
-
         // SUBROUTINE LOCAL VARIABLE DECLARATIONS:
         int NumOfSpeedCooling; // Number of speeds for cooling
         int NumOfSpeedHeating; // Number of speeds for heating
         int i;                 // Index to speed
 
-        auto &MSHeatPump = state.dataHVACMultiSpdHP->MSHeatPump;
+        auto &MSHeatPump = state.dataHVACMultiSpdHP->MSHeatPump(MSHeatPumpNum);
```
Comment (rraustad) [inline src/EnergyPlus/HVACMultiSpeedHeatPump.cc:2673]: [src/EnergyPlus/HVACMultiSpeedHeatPump.cc:2855]:(style),[constVariable],Variable 'MSHeatPump' can be declared as reference to const

File: src/EnergyPlus/HVACMultiSpeedHeatPump.cc [hunk 10]
```diff
@@ -3039,31 +2937,31 @@ namespace HVACMultiSpeedHeatPump {
         OnOffAirFlowRatio = 0.0;
         SupHeaterLoad = 0.0;
 
-        auto &MSHeatPump = state.dataHVACMultiSpdHP->MSHeatPump;
+        auto &MSHeatPump = state.dataHVACMultiSpdHP->MSHeatPump(MSHeatPumpNum);
```
Comment (rraustad) [inline src/EnergyPlus/HVACMultiSpeedHeatPump.cc:2853]: [src/EnergyPlus/HVACMultiSpeedHeatPump.cc:3042]:(style),[constVariable],Variable 'MSHeatPump' can be declared as reference to const

File: src/EnergyPlus/HVACMultiSpeedHeatPump.cc [hunk 11]
```diff
@@ -3146,11 +3044,11 @@ namespace HVACMultiSpeedHeatPump {
         // if the DX heating coil cannot meet the load, trim with supplemental heater
         // occurs with constant fan mode when compressor is on or off
         // occurs with cycling fan mode when compressor PLR is equal to 1
-        auto &MSHeatPump = state.dataHVACMultiSpdHP->MSHeatPump;
+        auto &MSHeatPump = state.dataHVACMultiSpdHP->MSHeatPump(MSHeatPumpNum);
```
Comment (rraustad) [inline src/EnergyPlus/HVACMultiSpeedHeatPump.cc:2960]: [src/EnergyPlus/HVACMultiSpeedHeatPump.cc:3149]:(style),[constVariable],Variable 'MSHeatPump' can be declared as reference to const

File: src/EnergyPlus/HVACMultiSpeedHeatPump.cc [hunk 12]
```diff
@@ -3245,10 +3137,10 @@ namespace HVACMultiSpeedHeatPump {
 
         OutsideDryBulbTemp = state.dataEnvrn->OutDryBulbTemp;
 
-        auto &MSHeatPump = state.dataHVACMultiSpdHP->MSHeatPump;
+        auto &MSHeatPump = state.dataHVACMultiSpdHP->MSHeatPump(MSHeatPumpNum);
```
Comment (rraustad) [inline src/EnergyPlus/HVACMultiSpeedHeatPump.cc:3053]: [src/EnergyPlus/HVACMultiSpeedHeatPump.cc:3248]:(style),[constVariable],Variable 'MSHeatPump' can be declared as reference to const

File: src/EnergyPlus/HVACMultiSpeedHeatPump.cc [hunk 13]
```diff
@@ -3834,73 +3718,42 @@ namespace HVACMultiSpeedHeatPump {
...
-        int FanInletNode;          // MSHP air outlet node
-        int FanOutletNode;         // MSHP air inlet node
         Real64 SavePartloadRatio;
         Real64 SaveSpeedRatio;
         Real64 QCoilActual;  // coil load actually delivered returned to calling component
         Real64 MinWaterFlow; // minimum water flow rate
         Real64 ErrorToler;   // supplemental heating coil convergence tollerance
 
-        auto &MSHeatPump = state.dataHVACMultiSpdHP->MSHeatPump;
+        auto &MSHeatPump = state.dataHVACMultiSpdHP->MSHeatPump(MSHeatPumpNum);
```
Comment (rraustad) [inline src/EnergyPlus/HVACMultiSpeedHeatPump.cc:3649]: [src/EnergyPlus/HVACMultiSpeedHeatPump.cc:3879]:(style),[constVariable],Variable 'MSHeatPump' can be declared as reference to const

File: src/EnergyPlus/HVACMultiSpeedHeatPump.cc [hunk 14]
```diff
@@ -4289,47 +3995,40 @@ namespace HVACMultiSpeedHeatPump {
...
         //  This routine will write values to output variables in MSHP
 
-        // METHODOLOGY EMPLOYED:
-
         Real64 TimeStepSysSec = state.dataHVACGlobal->TimeStepSysSec;
 
-        auto &MSHeatPump = state.dataHVACMultiSpdHP->MSHeatPump;
-        auto &MSHeatPumpReport = state.dataHVACMultiSpdHP->MSHeatPumpReport;
+        auto &MSHeatPump = state.dataHVACMultiSpdHP->MSHeatPump(MSHeatPumpNum);
+        auto &MSHeatPumpReport = state.dataHVACMultiSpdHP->MSHeatPumpReport(MSHeatPumpNum);
```
Comment (rraustad) [inline src/EnergyPlus/HVACMultiSpeedHeatPump.cc:3918]: [src/EnergyPlus/HVACMultiSpeedHeatPump.cc:4302]:(style),[constVariable],Variable 'MSHeatPump' can be declared as reference to const
[src/EnergyPlus/HVACMultiSpeedHeatPump.cc:4303]:(style),[constVariable],Variable 'MSHeatPumpReport' can be declared as reference to const

File: src/EnergyPlus/HVACMultiSpeedHeatPump.cc [hunk 15]
```diff
@@ -4342,48 +4041,36 @@ namespace HVACMultiSpeedHeatPump {
...
-        //       MODIFIED:        na
         //       RE-ENGINEERED    Revised to calculate MSHP heat recovery rate based on EIR Chiller heat recovery subroutine
         // PURPOSE OF THIS SUBROUTINE:
         //  Calculate the heat recovered from MSHP
 
-        // Using/Aliasing
-        auto &MSHPWasteHeat = state.dataHVACGlobal->MSHPWasteHeat;
-        using FluidProperties::GetSpecificHeatGlycol;
-        using PlantUtilities::SafeCopyPlantNode;
-
```
Comment (rraustad) [inline src/EnergyPlus/HVACMultiSpeedHeatPump.cc:4354]: [src/EnergyPlus/HVACMultiSpeedHeatPump.cc:4351]:(style),[constVariable],Variable 'MSHPWasteHeat' can be declared as reference to const

File: src/EnergyPlus/HVACMultiSpeedHeatPump.cc [hunk 16]
```diff
@@ -4597,52 +4258,54 @@ namespace HVACMultiSpeedHeatPump {
 
         QCoilActual = 0.0;
 
-        auto &MSHeatPump = state.dataHVACMultiSpdHP->MSHeatPump;
+        auto &MSHeatPump = state.dataHVACMultiSpdHP->MSHeatPump(MSHeatPumpNum);
```
Comment (rraustad) [inline src/EnergyPlus/HVACMultiSpeedHeatPump.cc:4174]: [src/EnergyPlus/HVACMultiSpeedHeatPump.cc:4600]:(style),[constVariable],Variable 'MSHeatPump' can be declared as reference to const

File: src/EnergyPlus/HVACMultiSpeedHeatPump.cc [hunk 17]
```diff
@@ -1944,80 +1874,54 @@ namespace HVACMultiSpeedHeatPump {
...
-        int i;                 // Index to speed
-        int NumOfSpeedCooling; // Number of speeds for cooling
-        int NumOfSpeedHeating; // Number of speeds for heating
-        Real64 DeltaMassRate;  // Difference of mass flow rate between inlet node and system outlet node
+        int i;                // Index to speed
+        Real64 DeltaMassRate; // Difference of mass flow rate between inlet node and system outlet node
 
-        int ZoneInSysIndex(0);                            // number of zone inlet nodes counter in an airloop
         int NumAirLoopZones(0);                           // number of zone inlet nodes in an air loop
-        int ZoneInletNodeNum(0);                          // zone inlet nodes node number
```
Comment (rraustad) [inline src/EnergyPlus/HVACMultiSpeedHeatPump.cc:1981]: [src/EnergyPlus/HVACMultiSpeedHeatPump.cc:1979]:(style),[unreadVariable],Variable 'ZoneInSysIndex' is assigned a value that is never used.
[src/EnergyPlus/HVACMultiSpeedHeatPump.cc:1981]:(style),[unreadVariable],Variable 'ZoneInletNodeNum' is assigned a value that is never used.

File: src/EnergyPlus/HVACMultiSpeedHeatPump.cc [hunk 18]
```diff
@@ -1944,80 +1874,54 @@ namespace HVACMultiSpeedHeatPump {
...
-        bool errFlag;
-        Real64 rho;    // local fluid density
-        Real64 MdotHR; // local temporary for heat recovery fluid mass flow rate (kg/s)
+        Real64 rho;                                       // local fluid density
+        Real64 MdotHR;                                    // local temporary for heat recovery fluid mass flow rate (kg/s)
         Real64 ZoneLoadToCoolSPSequenced;
         Real64 ZoneLoadToHeatSPSequenced;
 
         bool ErrorsFound(false);        // flag returned from mining call
-        int SteamIndex(0);              // index of steam quality for steam heating coil
```
Comment (rraustad) [inline src/EnergyPlus/HVACMultiSpeedHeatPump.cc:1991]: [src/EnergyPlus/HVACMultiSpeedHeatPump.cc:1991]:(style),[unreadVariable],Variable 'SteamIndex' is assigned a value that is never used.

File: src/EnergyPlus/HVACMultiSpeedHeatPump.cc [hunk 19]
```diff
@@ -3834,73 +3718,42 @@ namespace HVACMultiSpeedHeatPump {
...
+                OutsideDryBulbTemp =
+                    state.dataZoneTempPredictorCorrector->zoneHeatBalance(state.dataDXCoils->DXCoil(MSHeatPump.DXCoolCoilIndex).SecZonePtr).ZT;
             } else {
                 OutsideDryBulbTemp = state.dataEnvrn->OutDryBulbTemp;
             }
         } else {
             OutsideDryBulbTemp = state.dataEnvrn->OutDryBulbTemp;
         }
-        FanOutletNode = MSHeatPump(MSHeatPumpNum).FanOutletNode;
-        FanInletNode = MSHeatPump(MSHeatPumpNum).FanInletNode;
```
Comment (rraustad) [inline src/EnergyPlus/HVACMultiSpeedHeatPump.cc:3903]: [src/EnergyPlus/HVACMultiSpeedHeatPump.cc:3902]:(style),[unreadVariable],Variable 'FanOutletNode' is assigned a value that is never used.
[src/EnergyPlus/HVACMultiSpeedHeatPump.cc:3903]:(style),[unreadVariable],Variable 'FanInletNode' is assigned a value that is never used.

File: src/EnergyPlus/HVACMultiSpeedHeatPump.cc [hunk 20]
```diff
@@ -619,7 +546,7 @@ namespace HVACMultiSpeedHeatPump {
             if (lAlphaBlanks(2)) {
                 MSHeatPump(MSHPNum).AvaiSchedPtr = ScheduleManager::ScheduleAlwaysOn;
```
Comment (amirroth) [inline src/EnergyPlus/HVACMultiSpeedHeatPump.cc]: `auto &thisMSHeatPump` would help here. Also there is an unnecessary `IsNameEmpty` above.

File: src/EnergyPlus/HVACMultiSpeedHeatPump.cc [hunk 21]
```diff
@@ -736,30 +663,30 @@ namespace HVACMultiSpeedHeatPump {
             MSHeatPump(MSHPNum).FanName = Alphas(7);
             if (UtilityRoutines::SameString(Alphas(6), "Fan:OnOff") || UtilityRoutines::SameString(Alphas(6), "Fan:ConstantVolume")) {
```
Comment (amirroth) [inline src/EnergyPlus/HVACMultiSpeedHeatPump.cc]: 🤦‍♂️

File: src/EnergyPlus/HeatRecovery.cc [hunk 22]
```diff
@@ -2150,7 +2148,7 @@ namespace HeatRecovery {
             //   (supply air stream bypass mass flow rate proportional to ControlFraction except when frost control is active)
             if (this->ControlToTemperatureSetPoint) {
                 if ((this->SupInTemp - this->SupOutTemp) != 0.0) {
-                    if ((this->SupInTemp < HXTempSetPoint && this->SupOutTemp > HXTempSetPoint) ||
+                    if ((this->SupInTemp<HXTempSetPoint &&this->SupOutTemp> HXTempSetPoint) ||
```
Comment (rraustad) [inline src/EnergyPlus/HeatRecovery.cc]: Shoot

File: src/EnergyPlus/HVACMultiSpeedHeatPump.cc [hunk 23]
```diff
@@ -718,102 +639,72 @@ namespace HVACMultiSpeedHeatPump {
...
-                                       state.dataHVACMultiSpdHP->CurrentModuleObject,
-                                       MSHeatPump(MSHPNum).Name,
-                                       cAlphaFields(6),
-                                       Alphas(6)));
-                ShowContinueError(state, "Valid choices are Fan:OnOff or Fan:ConstantVolume");
+                ShowSevereError(
+                    state,
+                    format(
+                        "{}, \"{}\", fan {} \"{}\" not found", state.dataHVACMultiSpdHP->CurrentModuleObject, thisMSHP.Name, Alphas(6), Alphas(7)));
+                ShowContinueError(state, "Correct this fan name using valid Fan:OnOff or Fan:ConstantVolume fan object names.");
```
Comment (rraustad) [inline src/EnergyPlus/HVACMultiSpeedHeatPump.cc:674]: New warning (1st from GetFanIndex call and 2nd from MSHP here):

    ** Severe  ** AirLoopHVAC:UnitaryHeatPump:AirToAir:MultiSpeed, GetFanIndex: Fan not found=SUPPLY FAN 1A
    ** Severe  ** AirLoopHVAC:UnitaryHeatPump:AirToAir:MultiSpeed, "DXAC HEAT PUMP 1", fan FAN:ONOFF "SUPPLY FAN 1A" not found
    **   ~~~   ** Correct this fan name using valid Fan:OnOff or Fan:ConstantVolume fan object names.

**Comment by Myoldmopar** [comment]
Another great set of cppcheck fixes, building and testing locally now.

**Comment by Myoldmopar** [comment]
Everything passes happily, CI is happy, merging.  Thanks @rraustad

**Pull request merged**